### PR TITLE
docs: add scaling Python UDF servers

### PR DESCRIPTION
**Pull request by wangrunji0408**
## Info

- **Description**

  - This PR adds a section to the Python UDF page explaining how to scale Python UDF servers.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/udf/udf-python.md [hunk 1]
```diff
@@ -195,3 +195,90 @@ SELECT * FROM series(10);
 8
 9
 ```
+
+## 6. Scale the UDF Server
+
+Due to the limitations of the Python interpreter's [Global Interpreter Lock (GIL)](https://realpython.com/python-gil/), the UDF server can only utilize a single CPU core when processing requests. If you find that the throughput of the UDF server is insufficient, consider scaling up the UDF server.
```
Comment (xxchan) [inline docs/sql/udf/udf-python.md]:
```suggestion
Due to the limitations of the Python interpreter's [Global Interpreter Lock (GIL)](https://realpython.com/python-gil/), the UDF server can only utilize a single CPU core when processing requests. If you find that the throughput of the UDF server is insufficient, consider scaling out the UDF server.
```
?

File: docs/sql/udf/udf-python.md [hunk 2]
```diff
@@ -195,3 +195,90 @@ SELECT * FROM series(10);
...
+    server.serve()
+
+if __name__ == "__main__":
+    """Start multiple servers listening on different ports."""
+    n = 4
+    with Pool(n) as p:
+        p.map(start_server, range(8816, 8816 + n))
+```
+
+Then, you can use a load balancer like Nginx to distribute requests to different UDF servers.
```
Comment (xxchan) [inline docs/sql/udf/udf-python.md]: I'm not sure if the guide below is necessary. 🤪

Comment (wangrunji0408) [reply docs/sql/udf/udf-python.md]: At least it is necessary for me. I have never played with Nginx before. 🤪

Comment (hengm3467) [reply docs/sql/udf/udf-python.md]: I have the same question. :)  Maybe we can take the Nginx-specific information out? As the details may change and we don't want to be obliged to update descriptions that are out of our control. What do you think?  @wangrunji0408

Comment (wangrunji0408) [reply docs/sql/udf/udf-python.md]: Okay. I'll remove this part.

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**